### PR TITLE
Handle Xcode 12.1 IDEDiagnosticActivityLogMessage

### LINKED ISSUE #3: Crashed At IDEDiagnosticActivityLogMessage
Traceback (most recent call last):
        23: from /Users/bluesjiang/Desktop/Creative/tbtools/lib/TBLint/test.rb:22:in `<main>'
        22: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog.rb:14:in `parse_file'
        21: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/parser.rb:15:in `parse!'
        20: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/parser.rb:26:in `parse_tokens!'
        19: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:71:in `object'
        18: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:20:in `block in deserializer_for'
        17: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:26:in `deserialize_instance_of'
        16: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:26:in `each'
        15: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:29:in `block in deserialize_instance_of'
        14: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `object_list'
        13: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `new'
        12: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `initialize'
        11: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:64:in `block in object_list'
        10: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:71:in `object'
         9: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:20:in `block in deserializer_for'
         8: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:26:in `deserialize_instance_of'
         7: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:26:in `each'
         6: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:29:in `block in deserialize_instance_of'
         5: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `object_list'
         4: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `new'
         3: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:63:in `initialize'
         2: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:64:in `block in object_list'
         1: from /Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/slf0/token.rb:71:in `object'
/Users/bluesjiang/.gem/ruby/2.6.0/gems/xcactivitylog-0.2.1/lib/xcactivitylog/parser.rb:16:in `deserializer_for': uninitialized constant XCActivityLog::IDEDiagnosticActivityLogMessage (NameError)
Did you mean?  XCActivityLog::IDEClangDiagnosticActivityLogMessage

### PR DESCRIPTION
**Pull request by thiagohmcruz**
- Handle `IDEDiagnosticActivityLogMessage` (error from Xcode 12.1 logs)
- Add instructions on how to handle new `.xcactivitylog` files

Resolves #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,14 @@ To install this gem onto your local machine, run `bundle exec rake install`. To
 
 Bug reports and pull requests are welcome on GitHub at https://github.com/segiddins/xcactivitylog. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.
 
+## New Xcode versions
```
Comment (gyfelton) [inline README.md:39]: niceee

File: lib/xcactivitylog/objects.rb [hunk 2]
```diff
@@ -258,6 +258,9 @@ class IDEActivityLogAnalyzerControlFlowStepMessage < IDEActivityLogAnalyzerStepM
   class IDEActivityLogAnalyzerWarningMessage < IDEActivityLogMessage
     attributes.freeze
   end
+  class IDEDiagnosticActivityLogMessage < IDEActivityLogMessage
```
Comment (gyfelton) [inline lib/xcactivitylog/objects.rb:262]: NIT: empty line above?
```suggestion

  class IDEDiagnosticActivityLogMessage < IDEActivityLogMessage
```

**Review by gyfelton — APPROVED** [review]

**Review by segiddins — APPROVED** [review]

File: spec/util/xcactivity_log_yaml_helper.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require 'bundler/setup'
+require 'yaml'
+
+# Helper to sanitize the generated YAML due to different results on
+# CI vs local
+module XCActivityLogYAMLHelper
+  def sanitized_yaml(parsed_log)
```
Comment (thiagohmcruz) [inline spec/util/xcactivity_log_yaml_helper.rb:9]: @segiddins @gyfelton `rake spec` was working locally and failing on CI. I noticed that the output of `YAML.dump(parsed)` in `xcactivitylog_spec.rb` was different for some reason. 

Maybe it's a macOS vs Linux thing? I ran locally with the same versions of everything (`ruby`, `bundler`, etc) and the dumped yaml files were still different

I added this to fix the issues that I've found right before generating the yaml, so now regarding of what each call to `YAML.dump(parsed)` is doing natively the output is the same. All yaml files in `specs/fixtures/xcactivitylog` had to be re-generated to adopt the new pattern.

Let me know if this makes sense or if there's a better way/something I'm missing here

Comment (segiddins) [reply spec/util/xcactivity_log_yaml_helper.rb:9]: :lolsob:

**Review by segiddins — APPROVED** [review]

**Pull request merged**